athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/handlers/CompositeHandlerTest.java

```diff
@@ -219,4 +227,71 @@ public void doPing()
         compositeHandler.handleRequest(allocator, req, new ByteArrayOutputStream(), objectMapper);
         verify(mockMetadataHandler, times(1)).doPing(any(PingRequest.class));
     }
+
+    @Test
+    public void checkBucketAuthZ()
```

Comment:
I dont think this test makes sense on CompositeHandler, I think it belongs on MetadataHandlerTest

Reply:
There's no MetadataHandlerTest. Other concrete extensions of MetadataHander (such as DynamoDBMetadataHandler) are not in the SDK package, making unit tests for a private method in base class difficult.

Reply:
ExampleMetadataHandlerTest in the sdk module

Reply:
Unfortunately, ExampleMetadataHandlerTest is in package "package com.amazonaws.athena.connector.lambda.examples", while MetadataHandler is in "package com.amazonaws.athena.connector.lambda.handlers". The only test that is in the same package as Metadatahandler is CompositeHandlerTest.

Reply:
I dont think the package matters, you'll setup a test case in ExampleMetadataHandlerTest which attempts to use a bad bucket. calling getSplits should then result in the error you expect.

Reply:
Please see the revisions: simple caching is added to bucket authz checking, checking is moved to before getSplits, unit tests are moved to ExampleMetadataHandlerTest.